From 44b07a4da4c179cf78f2e12b1814f0aeca7105a9 Mon Sep 17 00:00:00 2001
From: Piotr Maziarz <piotrx.maziarz@linux.intel.com>
Date: Wed, 17 Jun 2020 09:52:52 +0200
Subject: [PATCH 1197/1214] ASoC: Intel: Skylake: Fix interrupt handling

Bring back validating incorrect value and CL DMA

Change-Id: I6fb9d2efdf9ed999cc16ee80b401c903bd7741c8
Tracked-On: PKT-3636, OAM-91517
Signed-off-by: Piotr Maziarz <piotrx.maziarz@linux.intel.com>
---
 sound/soc/intel/skylake/skl-sst-dsp.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/sound/soc/intel/skylake/skl-sst-dsp.c b/sound/soc/intel/skylake/skl-sst-dsp.c
index a0a9513..c4a2a9d 100644
--- a/sound/soc/intel/skylake/skl-sst-dsp.c
+++ b/sound/soc/intel/skylake/skl-sst-dsp.c
@@ -351,6 +351,14 @@ irqreturn_t skl_dsp_sst_interrupt(int irq, void *dev_id)
 	val = sst_dsp_shim_read_unlocked(ctx, SKL_ADSP_REG_ADSPIS);
 	ctx->intr_status = val;
 
+	if (val == 0xffffffff)
+		goto end;
+
+	if (val & SKL_ADSPIS_CL_DMA) {
+		skl_cldma_int_disable(ctx);
+		result = IRQ_WAKE_THREAD;
+	}
+
 	if (!(val & SKL_ADSPIS_IPC))
 		goto end;
 
-- 
2.7.4

